plugins/splunk-on-call/src/components/Incident/Incidents.tsx

```diff
@@ -15,24 +15,50 @@
  */
 
 import React, { useEffect } from 'react';
-import { List, ListSubheader } from '@material-ui/core';
+import {
+  createStyles,
+  List,
+  ListSubheader,
+  makeStyles,
+  Theme,
+} from '@material-ui/core';
 import { IncidentListItem } from './IncidentListItem';
 import { IncidentsEmptyState } from './IncidentEmptyState';
 import { useAsyncFn } from 'react-use';
 import { splunkOnCallApiRef } from '../../api';
 import { useApi, Progress } from '@backstage/core';
 import { Alert } from '@material-ui/lab';
 
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      maxHeight: '400px',
+      overflow: 'auto',
+    },
+    subheader: {
+      backgroundColor: 'white',
+    },
+    progress: {
+      margin: `0 ${theme.spacing(2)}px`,
+    },
+  }),
+);
+
 type Props = {
   refreshIncidents: boolean;
   team: string;
 };
 
 export const Incidents = ({ refreshIncidents, team }: Props) => {
+  const classes = useStyles();
   const api = useApi(splunkOnCallApiRef);
 
   const [{ value: incidents, loading, error }, getIncidents] = useAsyncFn(
     async () => {
+      // For some reason the changes applied to incidents (trigger-resolve-acknowledge)
+      // may take some time to actually be applied after receiving the response from the server.
+      // The timeout compensates for this latency.
+      await new Promise(resolve => setTimeout(resolve, 2000));
```

Comment:
Oh no 😅 , don't have a better suggestion that doesn't complicate things a lot more tho :p

Reply:
This method doesn't really suit me 😅  
Maybe we could use polling to check every `n-time` if we have the new status of the incidents.

I am open to other suggestions 👍